api/provider/aws/service_create.go

```diff
@@ -75,21 +75,53 @@ func (s *ServiceProvider) Create(req models.CreateServiceRequest) (string, error
 			securityGroupIDs = append(securityGroupIDs, loadBalancerSecurityGroup.GroupId)
 		}
 
-		for _, containerDefinition := range taskDefinition.ContainerDefinitions {
-			for _, portMapping := range containerDefinition.PortMappings {
-				for _, listenerDescription := range loadBalancerDescription.ListenerDescriptions {
-					listener := listenerDescription.Listener
-					if aws.Int64Value(listener.InstancePort) == aws.Int64Value(portMapping.ContainerPort) {
-						loadBalancer = &ecs.LoadBalancer{
-							ContainerName:    containerDefinition.Name,
-							ContainerPort:    portMapping.ContainerPort,
-							LoadBalancerName: loadBalancerDescription.LoadBalancerName,
+		var assignLoadBalancer = func() error {
+			for _, containerDefinition := range taskDefinition.ContainerDefinitions {
+				for _, portMapping := range containerDefinition.PortMappings {
+					var loadBalancerName *string
+					var targetGroupArn *string
+
+					// todo: check if this verifying this is actually needed
+					// why do we need to verify at least one container exposes a port that the elb has a listener for?
```

Comment:
Because if there are no exposed ports that tie to a listener, then the loadbalancer can't loadbalance anything. I think this was us trying to 'help'.

Reply:
Would you say this level of help is actually needed? Vanilla ECS doesn't enforce this and it is not like we were returning a useful error message when this happened. It looks like we would just silently fail.